### ISSUE TITLE
[BUG] Playwright immediately invokes function assignments 

### ISSUE THREAD

**Issue by straker** [microsoft/playwright-java#1070]
**Context:**
- Playwright Version: 1.25.0
- Operating System: Mac
- Browser: Chromium (not sure about others)

**Code Snippet**

```java
import com.microsoft.playwright.*;

public class ExampleReproducible {
  public static void main(String[] args) {
    try (Playwright playwright = Playwright.create()) {
      Browser browser = playwright.chromium().launch();
      BrowserContext context = browser.newContext();
      Page page = context.newPage();
      page.navigate("https://www.google.com/");
      page.evaluate("window.foo = () => { throw new Error('I ran') }");
    }
  }
}
```

**Describe the bug**

When evaluating code that assigns a function to a variable, Playwright Java is invoking the function at the same time. In the above example, this causes the Error to be thrown in the Java context instead of assigning `window.foo` to a function for later. As a comparison, in Playwright JavaScript this same evaluate would only create the assignment and not invoke the function.

What I would expect is that Playwright Java immediately invokes anonymous functions in order to pass variables into the browser scope but not invoke functions from an assignment. In order to assign a function for later what I end up having to doing is wrap the assignment in an anonymous function itself:

```java
page.evaluate("() => { window.foo = () => { throw new Error('I ran') } }");
```

**Comment by yury-s** [comment]
The behavior indeed differs between node.js and other language port of playwright. The reason for that is that there is no native syntax to define a function to be evaluated in the page and the expression is always passed as a string. Then in the page after the expression is evaluated if it results in a `function` object that function object is automatically invoked. Applying this to the example above, both  "() => { throw new Error('I ran') }" and "window.foo = () => { throw new Error('I ran') }" evaluate to the same function and hence it is automatically invoked. I will update our documentation to clarify this.
